OpenStack-Rabbit-Consumer/usr/local/bin/aq_api.py

```diff
@@ -32,9 +40,32 @@ def verify_kerberos_ticket():
     logger.info("Kerberos ticket success")
     return True
 
+def setup_requests(url, method, desc):
+    verify_kerberos_ticket()
 
-def aq_make(hostname, personality=None, osversion=None, archetype=None, osname=None):
-    logger.info("Attempting to make templates for " + hostname)
+    s = requests.Session()
+    s.verify = "/etc/grid-security/certificates/"
+    retries = Retry(total=5, backoff_factor=0.1, status_forcelist=[503])
+    s.mount('https://', HTTPAdapter(max_retries=retries))
+    if method == "post":
+        response = s.post(url, auth=HTTPKerberosAuth())
+    elif method == "put":
+        response = s.put(url, auth=HTTPKerberosAuth())
+    elif method == "delete":
+        response = s.delete(url, auth=HTTPKerberosAuth())
+    else:
+        response = s.get(url, auth=HTTPKerberosAuth())
+    if response.status_code != 200:
+        logger.error("%s: Failed: %s", desc, response.text)
+        logger.error(url)
+        raise Exception("%s: Failed", desc)
+
+    logger.info("%s: Success ", desc)
+    return response.text
+
+def aq_make(hostname, personality=None, osversion=None,
```

Comment:
The function arguments don't appear to be used directly in the function body, if they are optional/changeable would passing a `dict` make more sense here?